src/Data/Models/MRIUploadDTO.php

```diff
@@ -0,0 +1,182 @@
+     * @param string|null $upload_location             The UploadLocation
+     * @param string|null $decompress_location         The DecompressedLocation
+     * @param bool|null   $insertion_complete          The InsertionComplete
+     * @param bool|null   $inserting                   The Inserting
+     * @param string|null $patient_name                The PatientName
+     * @param int|null    $inserted_minc_count         The number_of_mincInserted
+     * @param int|null    $created_minc_count          The number_of_mincCreated
+     * @param int|null    $tarchive_id                 The TarchiveID
+     * @param int|null    $session_id                  The SessionID
+     * @param bool|null   $is_candidate_info_validated The IsCandidateInfoValidated
+     * @param bool|null   $is_tarchive_validated       The IsTarchiveValidated
+     * @param bool|null   $is_phantom                  The IsPhantom
+     */
+    public function __construct(
+        ?int $mri_upload_id,
+        ?string $uploaded_by,
+        \DateTime $upload_date,
+        ?string $upload_location,
+        ?string $decompress_location,
+        ?bool $insertion_complete,
+        ?bool $inserting,
+        ?string $patient_name,
+        ?int $inserted_minc_count,
+        ?int $created_minc_count,
+        ?int $tarchive_id,
+        ?int $session_id,
+        ?bool $is_candidate_info_validated,
+        ?bool $is_tarchive_validated,
+        ?bool $is_phantom
+    ) {
+        $this->mri_upload_id       = $mri_upload_id;
+        $this->uploaded_by         = $uploaded_by;
+        $this->upload_date         = $upload_date;
+        $this->upload_location     = $upload_location;
+        $this->decompress_location = $decompress_location;
+        $this->insertion_complete  = $insertion_complete ?? false;
+        $this->inserting           = $inserting;
+        $this->patient_name        = $patient_name;
+        $this->inserted_minc_count = $inserted_minc_count;
+        $this->created_minc_count  = $created_minc_count;
+        $this->tarchive_id         = $tarchive_id;
+        $this->session_id          = $session_id;
+        $this->is_candidate_info_validated = $is_candidate_info_validated;
+        $this->is_tarchive_validated       = $is_tarchive_validated ?? false;
+        $this->is_phantom = $is_phantom ?? false;
+    }
+
+    /**
+     * Helper function to instanciate an MRIUploadDTO object from a row in the
+     * database .
+     *
+     * @param array $row A row from the mri_upload table.
+     *
+     * @return MRIUploadDTO
+     */
+    public static function fromDBRow(array $row): MRIUploadDTO
+    {
+        $nullOrBool = function ($val) {
+            return is_null($val) ? null : boolval($val);
+        };
+
+        $nullOrInt = function ($val) {
+            return is_null($val) ? null : intval($val);
+        };
+
+        return new MRIUploadDTO(
+            intval($row['mri_upload_id']),
```

Comment:
@xlecours is intval needed here? The `mri_upload_id` value in the row should be an int in the database.